tests/Operation/ListCollectionsFunctionalTest.php

```diff
@@ -34,6 +34,34 @@ public function testListCollectionsForNewlyCreatedDatabase()
         }
     }
 
+    public function testCollectionInfoGetters()
+    {
+        if (version_compare($this->getServerVersion(), '3.4.0', '<')) {
+            $this->markTestSkipped('idIndex and info are not supported');
+        }
+
+        $server = $this->getPrimaryServer();
+
+        $operation = new DropDatabase($this->getDatabaseName());
+        $operation->execute($server);
+
+        $insertOne = new InsertOne($this->getDatabaseName(), $this->getCollectionName(), ['x' => 1]);
+        $writeResult = $insertOne->execute($server);
+        $this->assertEquals(1, $writeResult->getInsertedCount());
+
+        $operation = new ListCollections($this->getDatabaseName(), ['filter' => ['name' => $this->getCollectionName()]]);
+        $collections = $operation->execute($server);
+
+        $this->assertInstanceOf('MongoDB\Model\CollectionInfoIterator', $collections);
+
+        foreach ($collections as $collection) {
+            $this->assertInstanceOf('MongoDB\Model\CollectionInfo', $collection);
```

Comment:
Would it make sense to assert the type of the return of ``getInfo()`` and ``getIdIndex()`` values to the documented "document" too, if they are not ``null``?

Reply:
Good point. The return type should actually be `array|object|null`, since that is used in place of `document`. I fixed that in the most recent commit. I don't think the tests usually check the type of the return if it's just `array` or `object`.

Reply:
It's possible that ``document`` was used to refer to the ``array|object`` combination before, so perhaps ``document|null`` was "expected". In any case, your [change](https://github.com/mongodb/mongo-php-library/pull/520/files#diff-3b734177ac5b3b9a25cc0a63acfa4989R89) now left out the ``|null`` part, which I'm not sure whether that's what you wanted to do.

Reply:
My bad, thanks for catching that!